docs/problems-with-v3.md

```diff
@@ -0,0 +1,34 @@
+# Problems with Gatsby Source WordPress v3 :broken_heart:
+
+`gatsby-source-wordpress@v3`, the last major version of this plugin, utilized the [WP REST API](https://developer.wordpress.org/rest-api/) for sourcing WP data into Gatsby. This worked but was problematic for a few reasons.
+
+- The inflexibility and lack of an enforced schema for this API made proper node caching technically possible but functionally impossible.
+- The REST API is largely untyped, and REST does not enforce typing even if it was.
+  - This means Gatsby needed to use [inference](https://www.gatsbyjs.org/docs/glossary/#inference) to automatically build GraphQL types for the data that was returned from WP. The big issue with this is: if all data of a type is removed from WP, your Gatsby site's GraphQL queries will break because no data means Gatsby can no longer can infer the data structure. This led to a lot of odd and counterintuitive issues and a poor developer/content creator experience.
+  - If we were to manually write out types for the entire core WP REST API so that we no longer needed to use inference, any custom WP REST plugins would not work and would require a lot of additional tedious and error prone work to support these plugins changing their data structure over time.
```

Comment:
This also means that extensions, such as ACF to REST API can (and do) expose fields to WP REST endpoints that without being formally registered to the WP REST API Schema, meaning that the only way to know what's possible to be returned is to fetch every post and analyze what was returned. 

If the data coming from the WP REST API is different and unpredictable for each Post (or other resource), it makes it very difficult (impossible?) for decoupled applications to programmatically make use of the data being returned.

WPGraphQL resolves this by declaring what's possible to be returned as part of the Schema. Consumers, including Gatsby, can know what is possible via introspection of the GraphQL Schema without fetching a single Post.